applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -542,6 +551,41 @@ impl wallet_server::Wallet for WalletGrpcServer {
         Ok(Response::new(GetTransactionInfoResponse { transactions }))
     }
 
+    async fn stream_transaction_events(
+        &self,
+        _request: tonic::Request<TransactionEventRequest>,
+    ) -> Result<Response<Self::StreamTransactionEventsStream>, Status> {
+        let (mut sender, receiver) = mpsc::channel(100);
+        task::spawn(async move {
+            let event_listener = &WALLET_EVENT_LISTENER.receiver;
+            loop {
+                let transaction_event = match event_listener.recv_timeout(Duration::from_secs(1)) {
+                    Ok(source) => TransactionEvent::try_from(source),
+                    Err(_) => continue,
+                };
+                if transaction_event.is_ok() {
+                    let response = TransactionEventResponse {
+                        transaction: Some(transaction_event.unwrap()),
+                    };
+                    match sender.send(Ok(response)).await {
+                        Ok(_) => (),
+                        Err(err) => {
+                            warn!(target: LOG_TARGET, "Error sending transaction via GRPC:  {}", err);
+                            match sender.send(Err(Status::unknown("Error sending data"))).await {
+                                Ok(_) => (),
+                                Err(send_err) => {
+                                    warn!(target: LOG_TARGET, "Error sending error to GRPC client: {}", send_err)
+                                },
+                            }
+                            return;
+                        },
```

Comment:
Can be tidied up. (Clippy should catch this)

```suggestion
if let Err(err) = sender.send(Ok(response)).await {
                        warn!(target: LOG_TARGET, "Error sending transaction via GRPC:  {}", err);
                        if let Err(send_err) = sender.send(Err(Status::unknown("Error sending data"))).await {
                            warn!(target: LOG_TARGET, "Error sending error to GRPC client: {}", send_err)
                        }
                        return;
                    }
```